Repository loading: update load from diff mechanism (#6871)
ocaml/opam: master_changes.md

  * Compute the list of available depexts on `opam update` [#6489 @arozovyk - fix #6461]
  * Update depexts availability repository state cache when running `opam update --depexts` [#6489 @arozovyk - fix #6461]
  * Display status message while loading system package availability during `opam update` [#6489 @arozovyk - fix #6461]

>   * When loading a repository from a diff, no longer apply updates of unrelated files [#XXX @rjbou]

Author: ```suggestion
  * When loading a repository from a diff, no longer apply updates of unrelated files [#6871 @rjbou - fix 6872]
```

  * `OpamSolution.get_depexts` remove no longer needed `recover` option that was used with `--depext-only` option  [#6489 @arozovyk]

## opam-repository

>   * `OpamRepositoryPath`: add `install_nv_dir` to retrieve information from a path if it is an extra files path [#XXX @rjbou]

Author: ```suggestion
  * `OpamRepositoryPath`: add `install_nv_dir` to retrieve information from a path if it is an extra files path [#6871 @rjbou]
```

  * `OpamCompat.Map.add_to_list`: was added [#6818 @dra27]
  * `OpamSystem`: add `is_dir_read_only` [#6489 @rjbou]
  * `OpamFilename`: add `is_dir_read_only` [#6489 @rjbou]

>   * `OpamFilename`: add `to_list` that deconstructs a `filename` path [#XXX @rjbou]
>   * `OpamFilename.Dir`: add `of_list` that constructs a `dirname` from a string path elements [#XXX @rjbou]

Author: ```suggestion
  * `OpamFilename`: add `to_list` that deconstructs a `filename` path [#6871 @rjbou]
  * `OpamFilename.Dir`: add `of_list` that constructs a `dirname` from a string path elements [#6871 @rjbou]
```
